sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -37,11 +37,42 @@ export async function sendRequest(
 ): Promise<HttpResponse> {
   const httpClient = customHttpClient ?? getCachedDefaultHttpsClient();
   const request = buildPipelineRequest(method, url, options);
+  if (options) {
```

Comment:
since `options` is passed to `buildPipelineRequest`, maybe that helper should take care of inspecting `options` and passing relevant properties through to `createPipelineRequest`?